libraries/resource_providers/api500/c7000/storage_pool_provider.rb

```diff
@@ -18,20 +18,19 @@ module C7000
       class StoragePoolProvider < API300::C7000::StoragePoolProvider
         include OneviewCookbook::RefreshActions::RequestRefresh
 
+        def refresh
+          load_storage_system
+          super
+        end
+
         def load_storage_system
           raise("Unspecified property: 'storage_system'. Please set it before attempting this action.") unless @new_resource.storage_system
-          data = {
-            hostname: @new_resource.storage_system,
-            name: @new_resource.storage_system
-          }
-          @item.set_storage_system(load_resource(:StorageSystem, data))
+          @item['storageSystemUri'] = load_resource(:StorageSystem, { hostname: @new_resource.storage_system, name: @new_resource.storage_system }, :uri)
```

Comment:
Why did you remove the helper method?

Reply:
This helper method is not found to resource of API500, it is present only to API200 and API300.

Reply:
Well, it should be there in the API500 in oneview-sdk.

Please open an issue there and one here to change this later.